packages/dds/tree/src/test/shared-tree/editing.spec.ts

```diff
@@ -3,135 +3,159 @@
-			const sequencer = new Sequencer();
-			const tree1 = TestTree.fromJson(["A", "B", "C", "D"]);
-			const tree2 = tree1.fork();
-			const tree3 = tree1.fork();
-			const tree4 = tree1.fork();
+			const tree = makeTree("A", "B", "C", "D");
+			const delAB = tree.fork();
+			const delCD = tree.fork();
+			const addX = tree.fork();
+			const addY = tree.fork();
 
 			// Make deletions in two steps to ensure that gap tracking handles comparing insertion places that
 			// were affected by different deletes.
-			const delAB = remove(tree1, 0, 2);
-			const delCD = remove(tree2, 2, 2);
-			const addX = insert(tree3, 1, "x");
-			const addY = insert(tree4, 3, "y");
+			remove(delAB, 0, 2);
+			remove(delCD, 2, 2);
+			insert(addX, 1, "x");
+			insert(addY, 3, "y");
 
-			const sequenced = sequencer.sequence([delAB, delCD, addX, addY]);
-			tree1.receive(sequenced);
-			tree2.receive(sequenced);
-			tree3.receive(sequenced);
-			tree4.receive(sequenced);
+			delAB.merge();
+			delCD.merge();
+			addX.merge();
+			addY.merge();
 
-			expectJsonTree([tree1, tree2, tree3, tree4], ["x", "y"]);
+			expectJsonTree(tree, ["x", "y"]);
 		});
 
 		it("can handle competing deletes", () => {
 			for (const index of [0, 1, 2, 3]) {
-				const sequencer = new Sequencer();
 				const startingState = ["A", "B", "C", "D"];
-				const tree1 = TestTree.fromJson(startingState);
-				const tree2 = tree1.fork();
-				const tree3 = tree1.fork();
-				const tree4 = tree1.fork();
+				const tree = makeTree(...startingState);
+				const tree1 = tree.fork();
+				const tree2 = tree.fork();
+				const tree3 = tree.fork();
+				const tree4 = tree.fork();
 
-				const del1 = remove(tree1, index, 1);
-				const del2 = remove(tree2, index, 1);
-				const del3 = remove(tree3, index, 1);
+				remove(tree1, index, 1);
+				remove(tree2, index, 1);
+				remove(tree3, index, 1);
 
-				const sequenced = sequencer.sequence([del1, del2, del3]);
-				tree1.receive(sequenced);
-				tree2.receive(sequenced);
-				tree3.receive(sequenced);
-				tree4.receive(sequenced);
+				tree1.merge();
+				tree2.merge();
+				tree3.merge();
+				tree4.pull();
 
 				const expected = [...startingState];
 				expected.splice(index, 1);
-				expectJsonTree([tree1, tree2, tree3, tree4], expected);
+				expectJsonTree([tree, tree4], expected);
 			}
 		});
 
 		it("can rebase local dependent inserts", () => {
-			const sequencer = new Sequencer();
-			const tree1 = TestTree.fromJson("y");
+			const tree1 = makeTree("y");
 			const tree2 = tree1.fork();
-
-			const x = insert(tree1, 0, "x");
-
-			const ac = insert(tree2, 1, "a", "c");
-			const b = insert(tree2, 2, "b");
-			expectJsonTree(tree2, ["y", "a", "b", "c"]);
-
-			// Get an anchor to node b
-			const cursor = tree2.forest.allocateCursor();
-			moveToDetachedField(tree2.forest, cursor);
-			cursor.enterNode(2);
-			assert.equal(cursor.value, "b");
-			const anchor = cursor.buildAnchor();
-			cursor.free();
```

Comment:
I don't want to lose the anchor testing part. Is there something preventing it here?

Reply:
if you create this anchor after inserting b in tree 2, thats the last edit on that branch, so you can't do anything with it since its only valid on that branch, and the next thing done on that branch is the merge which ends that branch's lifetime.

Note that the design for anchorset was intended that the implementation of the merge would use an anchor to track where to put B, and thus would test something similar. My understanding is that this logic has been duplicated, so it makes sense to want to test the other copy of it.

If you did want to test an anchor across a merge, you would need to make the anchor on a branch that is getting content merged into it.